ERCS/erc-7965.md

```diff
@@ -0,0 +1,184 @@
+
+The key benefits of this approach include:
+
+- **Trustless verification**: No external validators or multisigs required. Chains trust their own consensus mechanisms.
+- **Universal compatibility**: Works between chains with verifiable state relationships through shared settlement infrastructure or compatible proof systems.
+- **Flexible messaging patterns**: Supports both targeted and broadcast messaging through standardized semantics, enabling new classes of applications like oracles and intent settlement systems.
+- **Composability**: Full integration with existing [ERC-7786] infrastructure and tooling
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Broadcasting Semantics
+
+Gateways MUST interpret an empty string `""` in the receiver address of `sendMessage()` as broadcasting. This convention allows cryptographic proof gateways to support both targeted messaging (to specific receivers) and broadcast messaging (to any interested party) within the same interface.
+
+Receivers of broadcast messages SHOULD validate the source and authenticity of messages according to their own security requirements.
+
+### Cryptographic Proof Attributes
+
+This specification defines the following [ERC-7786] attributes for cryptographic proof messaging. Gateways MUST return true if `supportsAttribute` is called with the selector for supported attributes.
+
+#### `route((bytes,bytes,uint256)[])`
+
+Specifies the verification path from destination to source chain with corresponding proofs and version requirements. Each tuple contains a bytes-encoded address that enables verification of the next chain's state or transaction history, the cryptographic proof required for that verification step, and the expected version of the verification logic (0 means any version is acceptable).
+
+When a non-zero version is specified, gateways MUST reject messages if the route address does not support the exact required version (unless `0`). Route addresses SHOULD implement version querying mechanisms to enable compatibility checking.
+
+The route MUST form a valid path where each step represents a direct relationship between chains that enables state or transaction verification. For multi-hop scenarios, the route creates a chain of trust where each step verifies the next, ultimately establishing the authenticity of the source chain's state. Gateways MUST reject messages with invalid or incomplete proof data.
+
+```solidity
+abi.encodeWithSignature("route((bytes,bytes,uint256)[])", router, proof, version);
```

Comment:
according to the signature, there should be a single argument that is an array of tuples. However, this mentions 3 separate arguments (are they array?), which is inconsistent.